nixos/modules/services/networking/firewall.nix

```diff
@@ -493,6 +510,51 @@ in
         '';
     };
 
+    networking.firewall.extraAllowed = mkOption {
+      type = types.listOf (types.submodule { options = {
+           sourceAddress = mkOption {
+                type = types.nullOr types.string;
+                default = null;
+                example = "198.51.100.0";
+                description = "Source of the packets. According to the iptables manual, DNS host names might be used here, but their use is discouraged (in particular because this delegates the security rule to DNS servers). DNS host names are resolved only when the rule is loaded.";
+           };
+           destAddress = mkOption {
+                type = types.nullOr types.string;
+                default = null;
+                example = "198.51.100.0";
+                description = "Destination of the packets. Same comment as for the `sourceAddress` parameter.";
+           };
+           protocols = mkOption {
+                type = types.nullOr (types.listOf (types.enum ["tcp" "udp" "icmp"]));
+                default = null;
+                example = ["tcp"];
```

Comment:
`example = [ "tcp" ];` would be consistent with the rest of the module (this applies to e.g. `[ "tcp" "udp" "icmp" ]` as well).